fluent-bit: 1.8.9 -> 1.8.11

fpletz opened the pull request:
Added openssl to deps since it yields better performance. Unfortunately
mbedtls is still being built (but that's apparently intended).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

globin commented:
@GrahamcOfBorg eval

SuperSandro2000 commented on pkgs/tools/misc/fluent-bit/default.nix:
```diff
@@ -1,26 +1,33 @@
-{ lib, stdenv, fetchFromGitHub, cmake, flex, bison, systemd }:
+{ lib, stdenv, fetchFromGitHub, cmake, flex, bison, pkgconfig
```
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, flex, bison, pkg-config
```

SuperSandro2000 commented on pkgs/tools/misc/fluent-bit/default.nix:
```diff
@@ -1,26 +1,33 @@
   };
 
-  nativeBuildInputs = [ cmake flex bison ];
+  nativeBuildInputs = [ cmake flex bison pkgconfig ];
```
```suggestion
  nativeBuildInputs = [ cmake flex bison pkg-config ];
```

SuperSandro2000 commented on pkgs/tools/misc/fluent-bit/default.nix:
```diff
@@ -1,26 +1,33 @@
+  nativeBuildInputs = [ cmake flex bison pkgconfig ];
 
-  buildInputs = lib.optionals stdenv.isLinux [ systemd ];
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isLinux [ systemd ];
```
```suggestion
  buildInputs = [ openssl ]
    ++ lib.optionals stdenv.isLinux [ systemd ];
```
is openssl needed on darwin?

fpletz replied:
I don't have access to a darwin machine to test. It should at least build. I don't see code for Apple's Secure Transport: https://github.com/fluent/fluent-bit/tree/master/src/tls

We could restrict it to Linux since nobody is probably using this on Darwin anyway. :smiley:

globin commented on pkgs/tools/misc/fluent-bit/default.nix:
```diff
@@ -1,26 +1,34 @@
 
   cmakeFlags = [ "-DFLB_METRICS=ON" "-DFLB_HTTP_SERVER=ON" ];
 
   patches = lib.optionals stdenv.isDarwin [ ./fix-luajit-darwin.patch ];
```
According to the ofborg log this patch is already applied in the new version on darwin

fpletz committed:
fluent-bit: 1.8.9 -> 1.8.11

Added openssl to deps since it yields better performance. Unfortunately
mbedtls is still being built (but that's apparently intended).
